python/pyspark/sql/connect/column.py

```diff
@@ -543,6 +543,39 @@ def __repr__(self) -> str:
         return f"({self._col} ({self._data_type}))"
 
 
+class LambdaFunction(Expression):
+    def __init__(
+        self,
+        function: Expression,
+        arguments: Sequence[Expression],
+    ) -> None:
+        super().__init__()
+
+        assert isinstance(function, Expression)
+
+        assert (
+            isinstance(arguments, list)
+            and len(arguments) > 0
+            and all(isinstance(arg, ColumnReference) for arg in arguments)
+        )
```

Comment:
Adding these assertions here is helpful in the Python client but the server side does not do the same assertion. What happens if we drop the assertion on `ColumnReference` what would happen on the server? 

Is the analysis exception not better than the Python assertion>

Reply:
we also check `UnresolvedAttribute` in the server side